torch/csrc/profiler/standalone/execution_trace_observer.cpp

```diff
@@ -377,9 +382,39 @@ static ExecutionTraceObserver::ID getObjectID(
   return iter->second;
 }
 
+void dumpTensorData2File(
+    std::string& tensor_dump_file_name,
+    Device device,
+    const char* storage_data_ptr,
+    size_t tensor_offset,
+    size_t tensor_nbyte) {
+  std::fstream fs;
+  fs.open(tensor_dump_file_name, std::fstream::out | std::fstream::binary);
+  if (fs.is_open()) {
+    if (device.is_cpu()) {
+      fs.write(
+          (const char*)storage_data_ptr + tensor_offset, (long)tensor_nbyte);
+    } else {
+      std::vector<char> host_data(tensor_nbyte);
+      c10::CopyBytes(
+          tensor_nbyte,
+          storage_data_ptr,
+          device,
+          (void*)host_data.data(),
+          c10::Device(c10::DeviceType::CPU),
+          false);
+      fs.write(
+          (const char*)host_data.data() + tensor_offset, (long)tensor_nbyte);
```

Comment:
Why do you need cast there?
```suggestion
          host_data.data() + tensor_offset, (long)tensor_nbyte);
```

Reply:
fixed.